Update xsltUnicodeSortFunction()

ddkilzer opened the pull request:
#### 6649227d017b97265cd5722a2117637aa932abad
<pre>
Update xsltUnicodeSortFunction()
<a href="https://bugs.webkit.org/show_bug.cgi?id=247538">https://bugs.webkit.org/show_bug.cgi?id=247538</a>
&lt;rdar://101996640&gt;

Reviewed by Alex Christensen.

* Source/WebCore/xml/XSLTUnicodeSort.cpp:
(WebCore::xsltUnicodeSortFunction):
- Update with changes from examples/xsltICUSort.c in libxml2.

Canonical link: <a href="https://commits.webkit.org/256459@main">https://commits.webkit.org/256459@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/109a217353d7dfe1873ede4415b0031e6df2901c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95612 "4 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/4875 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28660 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105191 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/165482 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99598 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/4929 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33626 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/87984 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/101041 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101274 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/3610 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82228 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30669 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/85483 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87391 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73506 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39363 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/18942 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37063 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20248 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4461 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/41059 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/42896 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/43047 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39498 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/xml/XSLTUnicodeSort.cpp:
```diff
@@ -72,19 +72,18 @@ void xsltUnicodeSortFunction(xsltTransformContextPtr ctxt, xmlNodePtr *sorts, in
 #ifdef XSLT_REFACTORED
```
Do we still need this?

ddkilzer replied:
I left it in there since the original code uses it.  Do you feel strongly about it?

achristensen07 commented on Source/WebCore/xml/XSLTUnicodeSort.cpp:
```diff
@@ -101,41 +100,44 @@ void xsltUnicodeSortFunction(xsltTransformContextPtr ctxt, xmlNodePtr *sorts, in
+            number[j] = 0;
+            if (stype) {
+                if (xmlStrEqual(stype, reinterpret_cast<const xmlChar*>("text"))) {
+                    // number[j] already zero.
```
Maybe worth asserting?

ddkilzer replied:
No, you're missing the point of the empty block.

The code that would go here would be `number[j] = 0;`, but that's already done above, so this block has nothing to do other than to check for a valid value.  (I know, it's not WebKit style, but again, I didn't want to restructure this too much to make future changes easier to apply.)

Commit:
Update xsltUnicodeSortFunction()
https://bugs.webkit.org/show_bug.cgi?id=247538
<rdar://101996640>

Reviewed by Alex Christensen.

* Source/WebCore/xml/XSLTUnicodeSort.cpp:
(WebCore::xsltUnicodeSortFunction):
- Update with changes from examples/xsltICUSort.c in libxml2.

Canonical link: https://commits.webkit.org/256459@main
